spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3824,7 +3824,9 @@ def insert_text(event):
                     last_obj = getobj(text)
                     prev_char = text[-2] if len(text) > 1 else ''
                     if prev_char in {')', ']', '}'} or (last_obj and not last_obj.isdigit()):
-                        self._start_completion_timer()
+                        # Completions should be triggered immediately when an
+                        # autocompletion character is introduced.
+                        self.do_completion(automatic=True)

```

Comment:
So do you agree with this change @goanpeca? What about the problem you tried to solve with the timer before?

Reply:
Yes I agree, I do not really mind. The only question is:

is using `.` considered automatic completion? in which case, should disabling the option on the preferences disable also this?

Reply:
I think we shouldn't handle it as automatic

Reply:
@andfoy, so you say we should change `automatic=False` here?

Reply:
No @ccordoba12 that is a different thing. I mean that if you go to preferences and disable automatic completions then.

is `.` (dot completion) considered an automatic completion? and if not? is there a way to disable it if for some reason a user didi not want to have completions with a dot.

Reply:
> is . (dot completion) considered an automatic completion?

I'd say it is and this should be disabled accordingly.

Reply:
In that case please add that fix @andfoy since currently this is not disabled if automatic completions are disabled.

Reply:
Now that you mention it, dot and others are defined as autocompletion trigger characters per the LSP specs. So I don't know if we should disable those as part of the automatic completions?

Reply:
Maybe we should have aa different option for "token completions" @ccordoba12 ?

Reply:
Yes, I agree.